projects/exui-common-lib/src/lib/services/idle/idle.services.ts

```diff
@@ -0,0 +1,68 @@
+import { Injectable } from '@angular/core';
+import {DocumentInterruptSource, Idle, } from '@ng-idle/core';
+import {Keepalive} from '@ng-idle/keepalive';
+import {Observable, Subject} from 'rxjs';
+import {
+  delay,
+  distinctUntilChanged,
+  map
+} from 'rxjs/operators';
+import {IdleConfigModel} from '../../models/idle-config.model';
+
+@Injectable()
+export class IdleService {
+  private appStateEmitter: Subject<{type: string, countdown?: string; isVisible?: boolean}>;
```

Comment:
Also, why is this not an EventEmitter, but a Subject that has to be retrieved through a method? Ideally this would be a private EventEmitter, with not a method but an associated getter to access it.

Reply:
because eventEmmiter is a subject under the hood and decided to use subject.

Reply:
Still make it a getter at the least as you're accessing it as a property